mesa/datacollection.py

```diff
@@ -130,16 +135,31 @@ def _new_agent_reporter(self, name, reporter):
 
         Args:
             name: Name of the agent-level variable to collect.
-            reporter: Attribute string, or function object that returns the
-                      variable when given a model instance.
+            reporter: Attribute string, function object, method of a class/instance, or
+                      function with parameters placed in a list that returns the
+                      variable when given an agent instance.
         """
+        # Check if the reporter is an attribute string
         if isinstance(reporter, str):
             attribute_name = reporter
 
-            def reporter(agent):
+            def attr_reporter(agent):
                 return getattr(agent, attribute_name, None)
 
-            reporter.attribute_name = attribute_name
```

Comment:
This seems to be a bug introduced in https://github.com/projectmesa/mesa/commit/bc5a5cd949992fcaa750ab5c6a8fc1cdb3b0c968, and you fixed in this PR.